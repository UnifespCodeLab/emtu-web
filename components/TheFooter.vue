<template>
  <v-footer padless>
    <v-card flat tile width="100%" class="primary text-center">
      <v-card-text>
        <v-btn
          v-for="icon in icons"
          :key="icon.name"
          :href="icon.link"
          class="mx-4"
          target="_blank"
          icon
        >
          <v-icon color="white" size="24px">
            {{ icon.name }}
          </v-icon>
        </v-btn>
      </v-card-text>

      <v-divider />

      <v-card-text>
        <v-btn id="btn-solicitacao" color="white" to="/solicitacao" small text>
          NÃ£o encontrou sua rota? Nos informe aqui!
        </v-btn>
      </v-card-text>
    </v-card>
  </v-footer>
</template>

<script>
export default {
  name: 'TheFooter',
  data () {
    return {
      icons: [
        {
          name: 'mdi-linkedin',
          link: 'https://www.linkedin.com/company/codelab-unifesp/'
        },
        {
          name: 'mdi-email',
          link: 'mailto:codelab.unifesp@gmail.com?subject="link HTML"'
        },
        { name: 'mdi-github', link: 'https://github.com/UnifespCodeLab' }
      ]
    }
  }
}
</script>
