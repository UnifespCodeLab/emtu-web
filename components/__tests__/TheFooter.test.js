import { mount, createLocalVue, RouterLinkStub } from '@vue/test-utils'
import Vuetify from 'vuetify'

import TheFooter from '~/components/TheFooter.vue'

describe('Pages / TheFooter', () => {
  const localVue = createLocalVue()

  let vuetify
  let wrapper

  beforeEach(() => {
    vuetify = new Vuetify()

    wrapper = mount(TheFooter, {
      localVue,
      vuetify,
      stubs: {
        RouterLink: RouterLinkStub
      }
    })
  })

  describe('when the page render', () => {
    it('should matches the screenshot', () => {
      expect(wrapper.element).toMatchSnapshot()
    })

    it('should render buttons correctly', () => {
      const button = wrapper.findAllComponents('.v-btn')
      expect(button).toHaveLength(4)
    })

    it('should render linkedin icon correctly', () => {
      const logo = wrapper.find('.mdi-linkedin')
      expect(logo.exists()).toBe(true)
    })

    it('should render email icon correctly', () => {
      const logo = wrapper.find('.mdi-email')
      expect(logo.exists()).toBe(true)
    })

    it('should render github icon correctly', () => {
      const logo = wrapper.find('.mdi-github')
      expect(logo.exists()).toBe(true)
    })

    it('should render codelab icon correctly', () => {
      const logo = wrapper.find('#btn-codelab')
      expect(logo.exists()).toBe(true)
    })
  })
})
