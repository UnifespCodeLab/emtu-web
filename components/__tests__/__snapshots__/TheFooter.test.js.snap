// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pages / TheFooter when the page render should matches the screenshot 1`] = `
<footer
  class="v-footer v-sheet theme--light v-footer--padless"
>
  <div
    class="primary text-center v-card v-card--flat v-sheet theme--light rounded-0"
    style="width: 100%;"
  >
    <div
      class="v-card__text"
    >
      <a
        class="mx-4 v-btn v-btn--icon v-btn--round theme--light v-size--default"
        href="https://www.linkedin.com/company/codelab-unifesp/"
        target="_blank"
      >
        <span
          class="v-btn__content"
        >
          <i
            aria-hidden="true"
            class="v-icon notranslate mdi mdi-linkedin theme--light white--text"
            style="font-size: 24px;"
          />
        </span>
      </a>
      <a
        class="mx-4 v-btn v-btn--icon v-btn--round theme--light v-size--default"
        href="mailto:codelab.unifesp@gmail.com?subject=\\"link HTML\\""
        target="_blank"
      >
        <span
          class="v-btn__content"
        >
          <i
            aria-hidden="true"
            class="v-icon notranslate mdi mdi-email theme--light white--text"
            style="font-size: 24px;"
          />
        </span>
      </a>
      <a
        class="mx-4 v-btn v-btn--icon v-btn--round theme--light v-size--default"
        href="https://github.com/UnifespCodeLab"
        target="_blank"
      >
        <span
          class="v-btn__content"
        >
          <i
            aria-hidden="true"
            class="v-icon notranslate mdi mdi-github theme--light white--text"
            style="font-size: 24px;"
          />
        </span>
      </a>
    </div>
     
    <hr
      aria-orientation="horizontal"
      class="v-divider theme--light"
      role="separator"
    />
     
    <div
      class="v-card__text"
    >
      <a
        class="v-btn v-btn--router v-btn--text theme--light v-size--small white--text"
        id="btn-solicitacao"
      >
        <span
          class="v-btn__content"
        >
          
        NÃ£o encontrou sua rota? Nos informe aqui!
      
        </span>
      </a>
    </div>
  </div>
</footer>
`;
